frappe/public/js/frappe/form/form.js

```diff
@@ -403,11 +403,17 @@ frappe.ui.form.Form = class FrappeForm {
 			this.doc = frappe.get_doc(this.doctype, this.docname);
 
 			// check permissions
+			this.fetch_permissions();
 			if (!this.has_read_permission()) {
 				frappe.show_not_permitted(__(this.doctype) + " " + __(cstr(this.docname)));
 				return;
 			}
 
+			// update grids with new permissions
+			this.grids.forEach((table) => {
+				table.grid.refresh();
```

Comment:
Don't know any better way. Suggestions welcome.

Also why frm.grids contains tables and not grids? xD

Reply:
Table control can have `refresh` instead?